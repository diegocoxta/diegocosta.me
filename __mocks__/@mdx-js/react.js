module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
