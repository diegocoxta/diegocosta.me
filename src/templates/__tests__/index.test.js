import React from 'react';
import { render } from '@testing-library/react';

jest.mock('../../components/Header', () => () => <p>Header</p>);
jest.mock('../../components/Footer', () => () => <p>Footer</p>);
jest.mock('../../components/ArticleHeader', () => () => <p>ArticleHeader</p>);
jest.mock('../../components/Metatags', () => () => <p>Metatags</p>);
jest.mock('../../components/Search', () => () => <p>Search</p>);
jest.mock('../../components/Navigation', () => () => <p>Navigation</p>);
jest.mock('../../components/AboutMe', () => () => <p>AboutMe</p>);

import IndexPage from '../index';

describe('<IndexPage>', () => {
  it('should render properly', () => {
    const data = {
      articles: {
        edges: [
          {
            node: {
              frontmatter: {
                title: 'Title 1',
                date: '30/10/2020',
                tags: [],
                description: 'Awesome first article',
              },
              fields: {
                slug: '/path-to-article',
                readingTime: {
                  minutes: 5,
                },
              },
              excerpt: 'Awesome first article',
            },
          },
          {
            node: {
              frontmatter: {
                title: 'Title 2',
                date: '30/10/2020',
                tags: [],
                description: 'Awesome second article',
              },
              fields: {
                slug: '/path-to-article-2',
                readingTime: {
                  minutes: 2,
                },
              },
              excerpt: 'Awesome first article',
            },
          },
        ],
      },
    };

    const { baseElement, getAllByTestId, getByText } = render(<IndexPage data={data} />);
    expect(getAllByTestId('index-page-article').length).toEqual(2);
    expect(getByText('Awesome second article')).toBeTruthy();
    expect(getByText('Awesome first article')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
