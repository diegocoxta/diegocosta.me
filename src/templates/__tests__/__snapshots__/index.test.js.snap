// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndexTemplate> renders properly 1`] = `
.c0 {
  max-width: 800px;
  margin: 0 auto;
  padding: 0 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 40px 0 0 0;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c2 {
  margin: 0 5px;
  color: #0C0D0E;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 20px;
}

.c4 {
  background: linear-gradient(90deg,#FFFFFF 20px,transparent 1%) center, linear-gradient(#FFFFFF 20px,transparent 1%) center, #0C0D0E;
  background-size: 22px 22px;
  height: 120px;
  margin: 40px 0;
  width: 100%;
}

.c3 p {
  font-size: 18px;
  line-height: 1.4;
  color: #0C0D0E;
}

.c3 ul {
  margin: 0;
  padding: 0;
}

.c3 li {
  list-style: none;
  font-size: 18px;
  font-weight: 900;
  margin: 0 24px 5px 0;
}

.c3 li a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #0C0D0E;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  text-transform: uppercase;
}

.c3 li a:after {
  content: '.';
  display: block;
  color: #FFFFFF;
  font-size: 38px;
  line-height: 0.2;
}

.c3 li a:hover:after {
  color: #0C0D0E;
}

.c5 {
  margin: 0 0 60px 0;
  font-weight: 400;
  color: #0C0D0E;
}

.c6 {
  margin: 20px 0 0 0;
}

.c7 {
  font-size: 36px;
  margin: 0;
}

.c8 {
  color: #0C0D0E;
  box-shadow: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8:hover,
.c8:focus {
  border-bottom: 1px solid #0C0D0E;
  outline: none;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-bottom: 0;
  padding: 0;
}

.c9 {
  color: #737373;
  font-size: 18px;
  margin: 5px 0;
}

.c11 {
  line-height: 1.5;
}

.c11 a {
  color: #626D4A;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 1px solid #FFFFFF;
}

.c11 a:hover,
.c11 a:focus {
  border-bottom: 1px solid #626D4A;
  outline: none;
}

.c11 blockquote {
  border-left: 5px solid #626D4A;
  padding-left: 20px;
}

.c11 .gatsby-highlight {
  padding: 10px 0;
}

.c11 pre[class*='language-'],.c11 code[class*='language-'],
.c11 .gatsby-highlight {
  background-color: #FFFFFF;
  font-family: monospace;
  white-space: pre-wrap;
}

.c11 .line-numbers-rows,
.c11 .line-numbers-rows span::before {
  border: 0.5px solid #FFFFFF;
}

.c11 img {
  box-shadow: none !important;
}

.c11 h3 {
  font-size: 28px;
}

@media (min-width:760px) {
  .c3 {
    padding-bottom: 40px;
  }
}

@media (min-width:760px) {
  .c3 p {
    font-size: 24px;
  }
}

@media (min-width:760px) {
  .c3 ul {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:760px) {
  .c3 li {
    font-size: 20px;
    margin: 0 24px 0 0;
  }
}

@media (min-width:760px) {
  .c3 li a:after {
    line-height: 0.3;
  }
}

<body>
  <div>
    <section
      class="c0"
    >
      <main>
        <div
          class="c1"
        >
          <a
            aria-label="languageswitcher.toggle languages.pt"
            class="c2"
            href="/about-us"
            language="pt"
          >
            /
            pt
          </a>
          <a
            aria-label="languageswitcher.toggle languages.en"
            class="c2"
            href="/about-us"
            language="en"
          >
            en
          </a>
          <a
            aria-label="languageswitcher.toggle languages.es"
            class="c2"
            href="/about-us"
            language="es"
          >
            es
          </a>
        </div>
      </main>
    </section>
    <p>
      Header
    </p>
    <main>
      <p>
        Metatags
      </p>
      <section
        class="c0"
      >
        <main>
          <div
            class="c3"
          />
        </main>
      </section>
      <div
        class="c4"
      />
      <p>
        Search
      </p>
      <section
        class="c0"
      >
        <main>
          <article
            class="c5"
            data-testid="article-item"
          >
            <header
              class="c6"
            >
              <h2
                class="c7"
              >
                <a
                  class="c8"
                  data-testid="article-header-custom-link"
                  href="/path-to-article"
                >
                  Title 1
                </a>
              </h2>
              <p
                class="c9"
              >
                16/01/2021
                 
                 · 5 article.minutes article.ofReading
                 
              </p>
              <ul
                class="c10"
                data-testid="article-header-tags"
              />
            </header>
            <section
              class="c11"
            >
              Awesome first article
            </section>
          </article>
        </main>
      </section>
      <section
        class="c0"
      >
        <main>
          <article
            class="c5"
            data-testid="article-item"
          >
            <header
              class="c6"
            >
              <h2
                class="c7"
              >
                <a
                  class="c8"
                  data-testid="article-header-custom-link"
                  href="/path-to-article-2"
                >
                  Title 2
                </a>
              </h2>
              <p
                class="c9"
              >
                18/01/2020
                 
                 · 2 article.minutes article.ofReading
                 
              </p>
              <ul
                class="c10"
                data-testid="article-header-tags"
              />
            </header>
            <section
              class="c11"
            >
              Awesome second article
            </section>
          </article>
        </main>
      </section>
    </main>
    <p>
      Footer
    </p>
  </div>
</body>
`;
